### PR TITLE
Add helm chart for virtual nodes

### PR DESCRIPTION
**Pull request by dmitsh**
Deploying of virtual nodes is a 2-steps process.
First we deploy the KWOK controller in the cluster. This is one-time operation.
Then we deploy virtual nodes. This operation can be repeated to change type and count of virtual nodes.
Refer the details in docs/deployment.md

### CODE REVIEW

File: charts/virtual-nodes/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+apiVersion: v2
+name: virtual-nodes
+description: A Helm chart for Kubernetes
```
Comment (lalitadithya) [inline charts/virtual-nodes/Chart.yaml:3]:
```suggestion
description: A Helm chart for deploying virtual GPU & CPU nodes in Kubernetes
```

File: charts/virtual-nodes/templates/node.yaml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    status: "False"
+    type: ReadonlyFilesystem
+  - message: Flannel is running on this node
+    reason: FlannelDeviceAvailable
+    status: "False"
+    type: FlannelNetworkDeviceProblem
+  - message: Dgx has /raid
+    reason: DgxRaidOk
+    status: "False"
+    type: DgxRaidProblem
```
Comment (lalitadithya) [inline charts/virtual-nodes/templates/node.yaml:130]: We can remove this condition it was supposed for internal use only on a specific set of cluster nodes and not a standard condition.

File: charts/virtual-nodes/templates/node.yaml [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+  - message: Node is in NW Topology CM or feature disabled
+    reason: NodeIsAdded
+    status: "False"
+    type: NodeNotInNWTopologyCM
```
Comment (lalitadithya) [inline charts/virtual-nodes/templates/node.yaml:134]: Same comment as earlier about this

File: charts/virtual-nodes/templates/node.yaml [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    nodeType: {{ $params.nodeType }}
+    {{- if $params.nodeInstance }}
+    nodeInstance: {{ $params.nodeInstance }}
+    {{- end }}
+    pod_gpu_size.static: dynamic
+    preempt_upon_cordon: "true"
+    type: kwok
+  name: virtual-{{ $node.type }}-{{.}}
+status:
+  conditions:
```
Comment (lalitadithya) [inline charts/virtual-nodes/templates/node.yaml:102]: More generally, I'm wondering if we should just add the conditions added by kubelet by default. Users can provide values for any additional conditions that they'd like to add, what do you think?

Comment (dmitsh) [reply charts/virtual-nodes/templates/node.yaml]: addressed, thanks!

File: charts/virtual-nodes/templates/node.yaml (hunk 3 above)
Comment (dmitsh) [reply charts/virtual-nodes/templates/node.yaml]: addressed

File: charts/virtual-nodes/templates/node.yaml (hunk 2 above)
Comment (dmitsh) [reply charts/virtual-nodes/templates/node.yaml]: addressed

File: charts/virtual-nodes/Chart.yaml (hunk 1 above)
Comment (dmitsh) [reply charts/virtual-nodes/Chart.yaml]: addressed

File: docs/deployment.md [hunk 5]
```diff
@@ -1,5 +1,49 @@
...
+Knavigator integrates with KWOK to simulate large clusters with hundreds or thousands of virtual nodes. This allows for the execution of experiments in a resource-efficient manner, without the need to run actual user workloads. The integration is facilitated through the API Server, which communicates with KWOK to manage the virtual nodes.
+
+To install the KWOK controller on a Kubernetes cluster, please follow [there instructions](https://kwok.sigs.k8s.io/docs/user/kwok-in-cluster/):
+
+```bash
+KWOK_REPO=kubernetes-sigs/kwok
+KWOK_LATEST_RELEASE="v0.5.2"
+
+kubectl apply -f "https://github.com/${KWOK_REPO}/releases/download/${KWOK_LATEST_RELEASE}/kwok.yaml"
+kubectl apply -f "https://github.com/${KWOK_REPO}/releases/download/${KWOK_LATEST_RELEASE}/stage-fast.yaml"
```
[the remark this answers is not in the document]
Comment (yuanchen8911) [reply docs/deployment.md:43]: `kwok` controller has to be starting before deploying stages.
```
kwok-controller-686cc8cfcc-pc4f6   1/1     Running   0            57s
```
 For example,
```
# Deploy and install kwok
kubectl apply -f https://github.com/kubernetes-sigs/kwok/releases/download/v0.5.2/kwok.yaml --wait
kubectl get pod -n kube-system | grep kwok

# Deploy pod stages
kubectl apply -f https://github.com/kubernetes-sigs/kwok/releases/download/v0.5.2/stage-fast.yaml
kubectl apply -f https://github.com/kubernetes-sigs/kwok/raw/main/kustomize/stage/pod/chaos/pod-init-container-running-failed.yaml
kubectl apply -f https://github.com/kubernetes-sigs/kwok/raw/main/kustomize/stage/pod/chaos/pod-container-running-failed.yaml

# Show all the stages
Kubectl get stages
```

[the remark this answers is not in the document]
Comment (dmitsh) [reply docs/deployment.md:43]: | kwok controller has to be running before deploying stages
Not necessarily. Stages are using CRD that deployed by kwok.yaml.
So, as long as kwok.yaml comes before stage-fast.yaml, we are good.
It also aligned with the [documentation](https://kwok.sigs.k8s.io/docs/user/kwok-in-cluster).

[the remark this answers is not in the document]
Comment (dmitsh) [reply docs/deployment.md]: I added deployment of the stages, thanks!

[the remark this answers is not in the document]
Comment (yuanchen8911) [reply docs/deployment.md]: You are right. The controller itself doesn't need to run, but the CRDs have to be created (`k apply -f kwok.yaml`).

File: charts/virtual-nodes/templates/nodes.yaml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+{{- range until $count }}
+---
+apiVersion: v1
+kind: Node
+metadata:
+  annotations:
+    node.alpha.kubernetes.io/ttl: "0"
+    kwok.x-k8s.io/node: fake
+    {{- range $key, $value := $node.annotations }}
+      {{ $key }}: {{ $value | quote }}
```
Comment (lalitadithya) [inline charts/virtual-nodes/templates/nodes.yaml:72]: I think the spacing needs to be similar to labels 

```suggestion
    {{ $key }}: {{ $value | quote }}
```

Comment (dmitsh) [reply charts/virtual-nodes/templates/nodes.yaml]: addressed, thanks!

**Review by lalitadithya — APPROVED** [review]

**Pull request merged**